ir/src/tests/variables.rs

```diff
@@ -165,14 +152,15 @@ fn err_ic_variables_vector_with_inlined_vector() {
         let d = [a[0], [3, 4]]
         enf clk' = d[0][0]";
 
-    parse(source).expect_err("Parsing failed");
+    expect_diagnostic(source, "expected one of");
```

Comment:
I decided against explicitly listing all of the tokens expected at this point, since there are many, and I'm not sure if the ordering is guaranteed to remain stable. Since the previous assertion was basically not checking what caused the error at all, this assertion at least catches the fact that an unexpected token error was raised.

That said, I'm happy to change the assertion to cover all the expected tokens if you think that's valuable here.